internal/common/vote/vote.go

```diff
@@ -0,0 +1,188 @@
+package vote
+
+//will be generated from client
+var voteForRule map[int][]int
+var voteForElect map[int][]int
+
+func GetVotesForRule(ruleID string, numOfIslands int) map[int][]int {
+	votesLayoutRule := make(map[int][]int)
+	i := 1
+	for {
+		votesLayoutRule[i] = voteForRule[i]
+		i++
+		if i > numOfIslands {
+			break
+		}
+	}
+	return votesLayoutRule
+}
+
+func VoteRule(ruleID string, numOfIslands int) (int, int, bool, map[int][]int) {
+
+	//Get votes from islands for the rule changes.
+	//var votesLayoutRule map[int][]int
+	votesLayoutRule := GetVotesForRule(ruleID, numOfIslands)
+
+	//Calculate results of each island.
+	resultsOfAllIslands := make(map[int][]int)
+	j := 1
+	for {
+		if votesLayoutRule[j][0] > votesLayoutRule[j][1] {
+			resultsOfAllIslands[j] = []int{1, 0}
+		}
+		if votesLayoutRule[j][0] <= votesLayoutRule[j][1] {
+			resultsOfAllIslands[j] = []int{0, 1}
+		}
+		j++
+		if j > numOfIslands {
+			break
+		}
+	}
+
+	//Announce voting result
+	numacc := 0
+	numrej := 0
+	var ans bool
+	for _, v := range resultsOfAllIslands {
+		if v[0] == 1 {
+			numacc++
+		}
+		if v[1] == 1 {
+			numrej++
+		}
+	}
+	if numacc > numrej {
+		ans = true
+	}
+	if numacc <= numrej {
+		ans = false
+	}
+	return numacc, numrej, ans, votesLayoutRule
+}
+
+func GetVotesForElect(numOfIslands int) map[int][]int {
+	votesLayoutElect := make(map[int][]int)
+	i := 1
+	for {
+		votesLayoutElect[i] = voteForElect[i]
+		i++
+		if i > numOfIslands {
+			break
+		}
+	}
+
+	return votesLayoutElect
+}
+
+func VoteElect(numOfIslands int) (int, []int, map[int][]int) {
```

Comment:
The functions should all have comments defining what they actually do. Is this the function that allows the islands to cast a vote?

Reply:
No, this function is just to collect votes from every island and write them in a map. IIGO will send a voting request to islands to my understanding. I think not interacting with other parts of infrastructure is mainly because the only input needed for voting is every island's votes, which is not given by design team at this stage. Once the voting rules are done by design, I guess I could implement it then? Also from my understanding, connection between IIGO and voting means that functions in voting could be called in IIGO, but voting doesn't need any information from IIGO right?

Reply:
Sorry func VoteElect is to calculate the voting result(who is the winner), func GetVotesForElect is to collect votes from every island and write them in a map, same as VoteRule and GetVotesForRule, I'll add explanation in the code.